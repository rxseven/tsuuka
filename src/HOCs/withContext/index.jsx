import React from 'react';

import { Context } from 'components/core/skeleton/Providers/Global';

function withContext(WrappedComponent) {
  return function HOC(props) {
    return (
      <Context.Consumer>
        {context => <WrappedComponent {...props} context={context} />}
      </Context.Consumer>
    );
  };
}

export default withContext;
