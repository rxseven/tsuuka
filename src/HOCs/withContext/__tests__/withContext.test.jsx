import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import withContext from '../index';

describe('withContext HOC', () => {
  // Arrange
  const component = <withContext />;

  it('should render without crashing', () => {
    // Act
    const wrapper = shallow(component);

    // Assertions
    expect(wrapper).toBeDefined();
  });

  describe('Snapshot tests', () => {
    it('should render correctly', () => {
      // Act
      const tree = renderer.create(component).toJSON();

      // Assert
      expect(tree).toMatchSnapshot();
    });
  });
});
