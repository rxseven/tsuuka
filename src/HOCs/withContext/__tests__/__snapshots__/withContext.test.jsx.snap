// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withContext HOC Snapshot tests should render correctly 1`] = `<withContext />`;
